build/source/engine/allocspace.f90

```diff
@@ -107,35 +109,55 @@ subroutine allocate_gru_struc(nGRU,nHRU,err,message)
  if(err/=0)then; message=trim(message)//trim(cmessage); return; endif
 
  ! get gru_ix dimension length
- err = nf90_inq_dimid(ncid, "ngru", gruDimID);             if(err/=0)then; message=trim(message)//'problem finding nGRU dimension'; return; endif
- err = nf90_inquire_dimension(ncid, gruDimID, len = nGRU); if(err/=0)then; message=trim(message)//'problem reading nGRU dimension'; return; endif
-
- ! get hru_dim dimension length (ie., the number of hrus in entire domain)
- err = nf90_inq_dimid(ncid, "nhru", hruDimID);             if(err/=0)then; message=trim(message)//'problem finding nHRU dimension'; return; endif
- err = nf90_inquire_dimension(ncid, hruDimID, len = nHRU); if(err/=0)then; message=trim(message)//'problem reading nHRU dimension'; return; endif
+ err = nf90_inq_dimid(ncid, "gru", gruDimID);                if(err/=nf90_noerr)then; message=trim(message)//'problem finding gru dimension/'//trim(nf90_strerror(err)); return; endif
```

Comment:
Rules for some dimension and variable names?
HRU -- **`hru`**, `nhru`, `nHRU`
GRU -- `gru`, `ngru`
HRU Index -- `hruIndex` or `hruId` or `hru_id`
GRU Index -- `gruIndex` or `gru_id`